src/main/java/com/nvidia/spark/rapids/jni/SparkResourceAdaptor.java

```diff
@@ -182,13 +184,21 @@ public void doneWaitingOnPool(long threadId) {
     doneWaitingOnPool(getHandle(), threadId);
   }
 
+  public void forceRetryOOM(long threadId, int numOOMs) {
+    forceRetryOOM(getHandle(), threadId, numOOMs, RmmSpark.OomInjectionType.CPU_OR_GPU.ordinal(), 0);
+  }
+
   /**
    * Force the thread with the given ID to throw a GpuRetryOOM on their next allocation attempt.
    * @param threadId the ID of the thread to throw the exception (not java thread id).
    * @param numOOMs the number of times the GpuRetryOOM should be thrown
    */
-  public void forceRetryOOM(long threadId, int numOOMs) {
-    forceRetryOOM(getHandle(), threadId, numOOMs);
+  public void forceRetryOOM(long threadId, int numOOMs, int oomMode, int skipCount) {
+    assert numOOMs >= 0 : "non-negative numOoms expected: actual=" + numOOMs;
```

Comment:
nit: Would it be better to have these checks in the C++ code? If bad things can happen if we set these values wrong (like an infinite loop), then C++ would be better to avoid the possibility of ever setting them wrong even when assertions are disabled. But if nothing too bad is going to happen, then here is fine as this really is just test code anyways.

Reply:
I am doing it in C++ as well to catch stale `native` method signatures in Java.